streamlit_login.py

```diff
@@ -0,0 +1,486 @@
+    conn = get_connection()
+    cursor = get_cursor(conn)
+    try:
+        query = """INSERT INTO users (first_name, last_name, email_address, password)
+        VALUES (%s, %s, %s, %s);"""
+        cursor.execute(query, (first_name, last_name, new_email, new_password))
+        cursor.close()
+        conn.commit()
+        conn.close()
+        return True
+    except Exception as e:
+        st.error(f"Error inserting into the database (users): {e}")
+        return False
+
+
+def insert_into_website(website: str) -> int:
+    """Inserts new websites into the website table and returns the corresponding website id"""
+    try:
+        conn = get_connection()
+        cursor = get_cursor(conn)
+        if get_website_id(website):
+            return get_website_id(website)
+        else:
+            cursor.execute(
+                "INSERT INTO website (website_name) VALUES (%s);", (website,))
+            cursor.close()
+            conn.commit()
+            conn.close()
+            return get_website_id(website)
+    except Exception as e:
+        st.error(f"Error inserting into the database (website): {e}")
+        return None
+
+
+def insert_into_product(website_id: int, url: str) -> int:
+    """Inserts new products into the product table and returns the corresponding product id"""
+    product_info = scrape_from_html(get_html_from_url(url), url)
+    try:
+        conn = get_connection()
+        cursor = get_cursor(conn)
+        if get_product_id(url):
+            return get_product_id(url)
+        else:
+            cursor.execute(
+                """INSERT INTO product (product_name, url, website_id, original_price) VALUES (%s, %s, %s, %s);""",
+                (product_info.get("game_title"), url, website_id,
+                 clean_price(product_info.get("original_price")),))
+            cursor.close()
+            conn.commit()
+            conn.close()
+            return get_product_id(url)
+    except Exception as e:
+        st.error(f"Error inserting into the database (product): {e}")
+        return None
+
+
+def insert_into_subscription(user_id, product_id, notification_price):
+    """Inserts new subscriptions into the subscription table and returns the corresponding product id"""
+    try:
+        conn = get_connection()
+        cursor = get_cursor(conn)
+        if get_subscription_id(user_id, product_id):
+            return get_subscription_id(user_id, product_id)
+        else:
+            cursor.execute(
+                """INSERT INTO subscription (user_id, product_id, notification_price) VALUES (%s, %s, %s);""",
+                (user_id, product_id, notification_price,))
+            cursor.close()
+            conn.commit()
+            conn.close()
+            return get_subscription_id(user_id, product_id)
+    except Exception as e:
+        st.error(f"Error inserting into the database (subscription): {e}")
+        return None
+
+
+def insert_initial_price(price, product_id) -> None:
```

Comment:
Insert functions have a similar "DRY" related issue. May want to consider slimming the code down.